tcclient/http.go

```diff
@@ -91,8 +91,9 @@ func (connectionData *ConnectionData) APICall(payload interface{}, method, route
 				Hash: sha256.New,
 			}
 			reqAuth := hawk.NewRequestAuth(httpRequest, credentials, 0)
-			if connectionData.Credentials.Certificate != "" {
-				reqAuth.Ext = base64.StdEncoding.EncodeToString([]byte("{\"certificate\":" + connectionData.Credentials.Certificate + "}"))
+			reqAuth.Ext, err = getExtHeader(connectionData.Credentials)
+			if err != nil {
+				return nil, nil, fmt.Errorf("Internal error: was not able to generate hawk ext header from provided credentials: %q\n%s", connectionData.Credentials, err)
```

Comment:
To understand the nil, nil here see [Retry docs](https://godoc.org/github.com/taskcluster/httpbackoff#Retry) which is used on line 106 below.